solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -804,15 +807,29 @@ public BitDocSet getDocSetBits(Query q) throws IOException {
   }
 
   // only handle positive (non negative) queries
-  DocSet getPositiveDocSet(Query q) throws IOException {
-    DocSet answer;
-    if (filterCache != null) {
-      answer = filterCache.get(q);
-      if (answer != null) return answer;
+  DocSet getPositiveDocSet(Query query) throws IOException {
+    // TODO duplicated code with getDocSet, can maybe also use FutureDocSet here
+    boolean doCache = filterCache != null;
```

Comment:
I like this `doCache` local var for clarity but otherwise I don't see why you are making any changes to this method.  It also appears getDocSetNC might get called twice if there's a cache miss?

Reply:
Very possible that I had other changes here before and then I had to back some of them out due to failing tests